combine/combine.go

```diff
@@ -0,0 +1,426 @@
+/*******************************************************************************
+ * Copyright (c) 2021-2022 Genome Research Ltd.
+ *
+ * Author: Sendu Bala <sb10@sanger.ac.uk>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining
+ * a copy of this software and associated documentation files (the
+ * "Software"), to deal in the Software without restriction, including
+ * without limitation the rights to use, copy, modify, merge, publish,
+ * distribute, sublicense, and/or sell copies of the Software, and to
+ * permit persons to whom the Software is furnished to do so, subject to
+ * the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+ * CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ ******************************************************************************/
+
+package combine
+
+import (
+	"bufio"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strconv"
+	"strings"
+
+	"github.com/klauspost/pgzip"
+	"github.com/wtsi-ssg/wrstat/dgut"
+	"github.com/wtsi-ssg/wrstat/fs"
+)
+
+const bytesInMB = 1000000
+const pgzipWriterBlocksMultiplier = 2
+const combineDGUTOutputFileBasename = "combine.dgut.db"
+const numSummaryColumns = 2
+const numSummaryColumnsDGUT = 3
+const groupSumCols = 2
+const userGroupSumCols = 3
+const intBase = 10
+const dgutSumCols = 4
+const dgutStoreBatchSize = 100000
+
+const statDGUTSummaryOutputFileSuffix = ".dgut"
+const userOnlyPerm = 448
+
+// MergeAndOptionallyCompressFiles identifies files from the supplied suffix,
+// merges them, and then may or may not compress them based upon the type of
+// function supplied as mergeAndStreamFunc.
+func MergeAndOptionallyCompressFiles(sourceDir, inputFileSuffix, outputFileBasename string,
```

Comment:
This should be called "MergeInputFilesToOutputFile" or similar, and there's no need to mention compression. It's not concern of this function.